web-app-router/scenes/Portal/Teams/TeamId/Apps/AppId/Actions/ActionId/Danger/page/content/index.tsx

```diff
@@ -0,0 +1,96 @@
+"use client";
+import { DecoratedButton } from "@/components/DecoratedButton";
+import { DialogOverlay } from "@/components/DialogOverlay";
+import { DialogPanel } from "@/components/DialogPanel";
+import { Dialog } from "@/components/Dialog";
+import { useCallback, useState } from "react";
+import { CircleIconContainer } from "@/components/CircleIconContainer";
+import { AlertIcon } from "@/components/Icons/AlertIcon";
+import { useDeleteActionMutation } from "./graphql/client/delete-action.generated";
+import { usePathname, useRouter } from "next/navigation";
+import { toast } from "react-toastify";
+import { revalidatePath } from "next/cache";
+
+export const ActionDangerZoneContent = (props: { action: any }) => {
+  const { action } = props;
+  const [openDeleteModal, setOpenDeleteModal] = useState(false);
+  const router = useRouter();
+
+  const [deleteActionQuery, { loading: deleteActionLoading }] =
+    useDeleteActionMutation({});
+
+  const deleteAction = useCallback(async () => {
+    try {
+      const result = await deleteActionQuery({
+        variables: { id: action.id ?? "" },
+      });
+      if (result instanceof Error) {
+        throw result;
+      }
+      router.replace(`..?deleteAction=${action.action}`); // TODO: Need to refetch action list to update, waiting till we decide on state
```

Comment:
@andy-t-wang You can use `refetchQueries` to refetch list. [Example](https://github.com/worldcoin/developer-portal/blob/56371837d2967eb26827e88812a578c2d3269386/web/src/hooks/useKeys/index.ts#L91-L93)